mso/resource_mso_system_config.go

```diff
@@ -0,0 +1,239 @@
+	workflow := "disabled"
+	if changeControl["enable"].(bool) {
+		workflow = "enabled"
+	}
+	changeControlMap := map[string]interface{}{
+		"workflow":            workflow,
+		"number_of_approvers": strconv.Itoa(int(changeControl["numOfApprovers"].(float64))),
+	}
+	d.Set("change_control", changeControlMap)
+
+	return nil
+}
+
+func patchSystemConfig(d *schema.ResourceData, msoClient *client.Client, systemConfigId string) error {
+
+	var patchPayloads []models.Model
+
+	changeControl := d.Get("change_control")
+	if changeControl != nil {
+		changeControlMap := changeControl.(map[string]interface{})
+		enable := false
+		if changeControlMap["workflow"].(string) == "enabled" {
+			enable = true
+		}
+		approvers, err := strconv.Atoi(changeControlMap["number_of_approvers"].(string))
+		if err != nil {
+			return err
+		}
+		patchPayloads = append(patchPayloads, models.NewSystemConfigChangeControl(enable, approvers))
+	}
+
+	alias := d.Get("alias").(string)
+	banner := d.Get("banner")
+	if (banner != nil && len(banner.([]interface{})) > 0) || alias != "" {
+		bannerMap := banner.([]interface{})[0].(map[string]interface{})
+		patchPayloads = append(patchPayloads, models.NewSystemConfigBanner(
+			alias,
+			bannerMap["state"].(string),
+			bannerMap["type"].(string),
+			bannerMap["message"].(string),
+		))
+	}
+
+	if len(patchPayloads) > 0 {
+		_, err := msoClient.PatchbyID(fmt.Sprintf("%s/%s", systemConfigUrl, systemConfigId), patchPayloads...)
+		if err != nil {
+			log.Println(err)
+			return err
+		}
+	}
+
+	d.SetId(systemConfigId)
+
+	return nil
+}
+
+func resourceMSOSystemConfigCreate(d *schema.ResourceData, m interface{}) error {
+	log.Printf("[DEBUG] System Config Beginning Creation")
+	msoClient := m.(*client.Client)
+
+	con, err := msoClient.GetViaURL(systemConfigUrl)
+	if err != nil {
+		return err
+	}
+	systemConfigId := models.StripQuotes(con.Search("systemConfigs").Search("id").String())
+
+	err = patchSystemConfig(d, msoClient, systemConfigId)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] %s: System Config Creation finished successfully", d.Id())
+
+	return resourceMSOSystemConfigRead(d, m)
+}
+
+func resourceMSOSystemConfigUpdate(d *schema.ResourceData, m interface{}) error {
+	log.Printf("[DEBUG] Schema: Beginning Update")
+
+	err := patchSystemConfig(d, m.(*client.Client), d.Id())
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] %s: Schema Update finished successfully", d.Id())
+	return resourceMSOSystemConfigRead(d, m)
+}
+
+func resourceMSOSystemConfigRead(d *schema.ResourceData, m interface{}) error {
+	log.Printf("[DEBUG] %s: Beginning Read", d.Id())
+
+	err := getSystemConfig(d, m)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] %s: Read finished successfully", d.Id())
+	return nil
+}
+
+func resourceMSOSystemConfigDelete(d *schema.ResourceData, m interface{}) error {
+	log.Printf("[DEBUG] %s: Beginning Destroy", d.Id())
+	d.SetId("")
```

Comment:
I am confused, there's no API call to delete the config?

just set the id to "" ?

Reply:
Yes because there is also no create done. [See]( https://github.com/CiscoDevNet/terraform-provider-mso/pull/241/files/68c9da659361761625f33eaa4e001b2b78e036ab#diff-698807db8bb9e2852cdc0abf18b54e52ade1f3a8d37b4f00bc292c2da8dfcf7dR200) where generic PATCH function is called.  Then [see](https://github.com/CiscoDevNet/terraform-provider-mso/pull/241/files/68c9da659361761625f33eaa4e001b2b78e036ab#diff-698807db8bb9e2852cdc0abf18b54e52ade1f3a8d37b4f00bc292c2da8dfcf7dR162) and [see](https://github.com/CiscoDevNet/terraform-provider-mso/pull/241/files/68c9da659361761625f33eaa4e001b2b78e036ab#diff-698807db8bb9e2852cdc0abf18b54e52ade1f3a8d37b4f00bc292c2da8dfcf7dR169) where both patch payloads do not have a operation as we normally define. In the go client you can [see](https://github.com/ciscoecosystem/mso-go-client/blob/master/models/system_config.go) that "replace" is statically set. The reason for this is that the system config is always already present, and is not something we create or delete. 

I have considered that we should keep track of original state in computed value one thus that delete would reset to the state pre-creation, but have decided to just remove from state.